### ISSUE TITLE
Build failure: librewolf

### ISSUE THREAD

**Issue by Schweber** [NixOS/nixpkgs#336490]
### Steps To Reproduce

1. build *librewolf*

### Build log

```
warning: The interpretation of store paths arguments ending in `.drv` recently changed. If this command is now failing try again with '/nix/store/phkbrpwii0qb8x86nl765h0csw7f3na2-librewolf-unwrapped-129.0.1-1.drv^*'
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/yvfbccqmlyz22aygk7pg6s83lpx5nx4n-firefox-129.0.1.source.tar.xz
source root is firefox-129.0.1
setting SOURCE_DATE_EPOCH to timestamp 1723466949 of file firefox-129.0.1/sourcestamp.txt
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/9lx1746yy56pn3dpc35l1xr6l3lykbin-env_var_for_system_dir-ff111.patch
patching file toolkit/xre/nsXREDirProvider.cpp
Hunk #2 succeeded at 313 (offset 3 lines).
applying patch /nix/store/6wsdh6246xbmlv3jr5q7qpl9ay3acq04-no-buildconfig-ffx121.patch
patching file docshell/base/nsAboutRedirector.cpp
patching file toolkit/content/jar.mn
Hunk #1 succeeded at 44 (offset 3 lines).
applying patch /nix/store/14gmmaq28hb5892lvwap2yblww7afgpz-cbindgen-0.27.0-compat.patch
patching file servo/components/style_traits/values.rs
Hunk #1 succeeded at 383 (offset -1 lines).
Hunk #2 succeeded at 408 (offset -1 lines).
patching file servo/ports/geckolib/cbindgen.toml
Hunk #1 succeeded at 355 (offset -2 lines).
applying patch /nix/store/idkwbyzx5hv6cjbmq6aplxpkxyji0n4x-mozbz-1898476-1.patch
patching file widget/gtk/MozContainerWayland.cpp
patching file widget/gtk/MozContainerWayland.h
patching file widget/gtk/moz.build
patching file widget/gtk/MozContainerSurfaceLock.cpp
patching file widget/gtk/MozContainerSurfaceLock.h
applying patch /nix/store/n6drjva08mkxwrx0ikz9xkzby4zsfail-mozbz-1898476-2.patch
patching file widget/gtk/GtkCompositorWidget.cpp
patching file widget/gtk/GtkCompositorWidget.h
patching file widget/gtk/nsWindow.cpp
patching file widget/gtk/nsWindow.h
applying patch /nix/store/7f249yx91fri3wb2nwgy1k8l8ms4sqm9-mozbz-1898476-3.patch
patching file gfx/webrender_bindings/RenderCompositorEGL.cpp
patching script interpreter paths in mach build
mach: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/04gg5w1s662l329a8kh9xcwyp0k64v5a-python3-3.12.4/bin/python3"
build/cargo-linker: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/04gg5w1s662l329a8kh9xcwyp0k64v5a-python3-3.12.4/bin/python3"
build/vs/generate_yaml.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/04gg5w1s662l329a8kh9xcwyp0k64v5a-python3-3.12.4/bin/pyth>
build/checksums.py: interpreter directive changed from "#!/usr/bin/python" to "/nix/store/04gg5w1s662l329a8kh9xcwyp0k64v5a-python3-3.12.4/bin/python"
build/macosx/xcrun: interpreter directive changed from "#!/bin/sh" to "/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/sh"
build/cargo-host-linker: interpreter directive changed from "#!/bin/sh" to "/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/sh"
build/build-clang/build-clang.py: interpreter directive changed from "#!/usr/bin/python3" to "/nix/store/04gg5w1s662l329a8kh9xcwyp0k64v5a-python3-3.12.4/bin/p>
build/clang-plugin/import_mozilla_checks.py: interpreter directive changed from "#!/usr/bin/python3" to "/nix/store/04gg5w1s662l329a8kh9xcwyp0k64v5a-python3-3>
build/unix/print-non-newline.sh: interpreter directive changed from "#!/bin/sh" to "/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/sh"
build/unix/build-hfsplus/build-hfsplus.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/ba>
build/unix/build-binutils/build-binutils.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/>
build/unix/build-gcc/build-gcc.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/bash"
build/pgo/profileserver.py: interpreter directive changed from "#!/usr/bin/python" to "/nix/store/04gg5w1s662l329a8kh9xcwyp0k64v5a-python3-3.12.4/bin/python"
build/package/mac_osx/unpack-diskimage: interpreter directive changed from "#!/bin/bash" to "/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/bash"
build/package/mac_osx/make-diskimage: interpreter directive changed from "#!/bin/sh" to "/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/sh"
build/qemu-wrap: interpreter directive changed from "#!/bin/bash" to "/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/bash"
build/autoconf/config.guess: interpreter directive changed from "#! /bin/sh" to "/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/sh"
build/autoconf/install-sh: interpreter directive changed from "#!/bin/sh" to "/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/sh"
build/autoconf/config.sub: interpreter directive changed from "#! /bin/sh" to "/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/sh"
applying LibreWolf patch: patches/allow-JXL-in-non-nightly-browser.patch
patching file toolkit/moz.configure
Hunk #1 succeeded at 905 (offset 202 lines).
applying LibreWolf patch: patches/allow-ubo-private-mode.patch
patching file toolkit/components/extensions/Extension.sys.mjs
Hunk #1 succeeded at 3717 (offset 431 lines).
applying LibreWolf patch: patches/bootstrap.patch
patching file python/mozversioncontrol/mozversioncontrol/__init__.py
Hunk #1 succeeded at 1137 (offset 118 lines).
applying LibreWolf patch: patches/context-menu.patch
patching file browser/base/content/browser-context.inc
applying LibreWolf patch: patches/custom-ubo-assets-bootstrap-location.patch
patching file toolkit/components/extensions/parent/ext-storage.js
Hunk #1 succeeded at 344 (offset 167 lines).
applying LibreWolf patch: patches/dbus_name.patch
patching file toolkit/components/remote/nsDBusRemoteClient.cpp
Hunk #3 succeeded at 114 (offset -1 lines).
Hunk #4 succeeded at 125 (offset -1 lines).
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/yvfbccqmlyz22aygk7pg6s83lpx5nx4n-firefox-129.0.1.source.tar.xz
source root is firefox-129.0.1
setting SOURCE_DATE_EPOCH to timestamp 1723466949 of file firefox-129.0.1/sourcestamp.txt
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/9lx1746yy56pn3dpc35l1xr6l3lykbin-env_var_for_system_dir-ff111.patch
patching file toolkit/xre/nsXREDirProvider.cpp
Hunk #2 succeeded at 313 (offset 3 lines).
applying patch /nix/store/6wsdh6246xbmlv3jr5q7qpl9ay3acq04-no-buildconfig-ffx121.patch
patching file docshell/base/nsAboutRedirector.cpp
patching file toolkit/content/jar.mn
Hunk #1 succeeded at 44 (offset 3 lines).
applying patch /nix/store/14gmmaq28hb5892lvwap2yblww7afgpz-cbindgen-0.27.0-compat.patch
patching file servo/components/style_traits/values.rs
Hunk #1 succeeded at 383 (offset -1 lines).
Hunk #2 succeeded at 408 (offset -1 lines).
patching file servo/ports/geckolib/cbindgen.toml
Hunk #1 succeeded at 355 (offset -2 lines).
applying patch /nix/store/idkwbyzx5hv6cjbmq6aplxpkxyji0n4x-mozbz-1898476-1.patch
patching file widget/gtk/MozContainerWayland.cpp
patching file widget/gtk/MozContainerWayland.h
patching file widget/gtk/moz.build
patching file widget/gtk/MozContainerSurfaceLock.cpp
patching file widget/gtk/MozContainerSurfaceLock.h
applying patch /nix/store/n6drjva08mkxwrx0ikz9xkzby4zsfail-mozbz-1898476-2.patch
patching file widget/gtk/GtkCompositorWidget.cpp
patching file widget/gtk/GtkCompositorWidget.h
patching file widget/gtk/nsWindow.cpp
patching file widget/gtk/nsWindow.h
applying patch /nix/store/7f249yx91fri3wb2nwgy1k8l8ms4sqm9-mozbz-1898476-3.patch
patching file gfx/webrender_bindings/RenderCompositorEGL.cpp
patching script interpreter paths in mach build
mach: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/04gg5w1s662l329a8kh9xcwyp0k64v5a-python3-3.12.4/bin/python3"
build/cargo-linker: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/04gg5w1s662l329a8kh9xcwyp0k64v5a-python3-3.12.4/bin/python3"
build/vs/generate_yaml.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/04gg5w1s662l329a8kh9xcwyp0k64v5a-python3-3.12.4/bin/pyth>
build/checksums.py: interpreter directive changed from "#!/usr/bin/python" to "/nix/store/04gg5w1s662l329a8kh9xcwyp0k64v5a-python3-3.12.4/bin/python"
build/macosx/xcrun: interpreter directive changed from "#!/bin/sh" to "/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/sh"
build/cargo-host-linker: interpreter directive changed from "#!/bin/sh" to "/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/sh"
build/build-clang/build-clang.py: interpreter directive changed from "#!/usr/bin/python3" to "/nix/store/04gg5w1s662l329a8kh9xcwyp0k64v5a-python3-3.12.4/bin/p>
build/clang-plugin/import_mozilla_checks.py: interpreter directive changed from "#!/usr/bin/python3" to "/nix/store/04gg5w1s662l329a8kh9xcwyp0k64v5a-python3-3>
build/unix/print-non-newline.sh: interpreter directive changed from "#!/bin/sh" to "/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/sh"
build/unix/build-hfsplus/build-hfsplus.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/ba>
build/unix/build-binutils/build-binutils.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/>
build/unix/build-gcc/build-gcc.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/bash"
build/pgo/profileserver.py: interpreter directive changed from "#!/usr/bin/python" to "/nix/store/04gg5w1s662l329a8kh9xcwyp0k64v5a-python3-3.12.4/bin/python"
build/package/mac_osx/unpack-diskimage: interpreter directive changed from "#!/bin/bash" to "/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/bash"
build/package/mac_osx/make-diskimage: interpreter directive changed from "#!/bin/sh" to "/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/sh"
build/qemu-wrap: interpreter directive changed from "#!/bin/bash" to "/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/bash"
build/autoconf/config.guess: interpreter directive changed from "#! /bin/sh" to "/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/sh"
build/autoconf/install-sh: interpreter directive changed from "#!/bin/sh" to "/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/sh"
build/autoconf/config.sub: interpreter directive changed from "#! /bin/sh" to "/nix/store/4bj2kxdm1462fzcc2i2s4dn33g2angcc-bash-5.2p32/bin/sh"
applying LibreWolf patch: patches/allow-JXL-in-non-nightly-browser.patch
patching file toolkit/moz.configure
Hunk #1 succeeded at 905 (offset 202 lines).
applying LibreWolf patch: patches/allow-ubo-private-mode.patch
patching file toolkit/components/extensions/Extension.sys.mjs
Hunk #1 succeeded at 3717 (offset 431 lines).
applying LibreWolf patch: patches/bootstrap.patch
patching file python/mozversioncontrol/mozversioncontrol/__init__.py
Hunk #1 succeeded at 1137 (offset 118 lines).
applying LibreWolf patch: patches/context-menu.patch
patching file browser/base/content/browser-context.inc
applying LibreWolf patch: patches/custom-ubo-assets-bootstrap-location.patch
patching file toolkit/components/extensions/parent/ext-storage.js
Hunk #1 succeeded at 344 (offset 167 lines).
applying LibreWolf patch: patches/dbus_name.patch
patching file toolkit/components/remote/nsDBusRemoteClient.cpp
Hunk #3 succeeded at 114 (offset -1 lines).
Hunk #4 succeeded at 125 (offset -1 lines).
patching file toolkit/components/remote/nsDBusRemoteServer.cpp
Hunk #1 succeeded at 29 (offset 1 line).
Hunk #2 succeeded at 39 (offset 1 line).
Hunk #3 succeeded at 130 (offset 1 line).
Hunk #4 succeeded at 204 (offset 2 lines).
Hunk #5 succeeded at 218 (offset 2 lines).
applying LibreWolf patch: patches/devtools-bypass.patch
patching file devtools/server/actors/thread.js
patching file devtools/server/actors/webconsole/listeners/console-api.js
applying LibreWolf patch: patches/disable-data-reporting-at-compile-time.patch
patching file browser/moz.configure
patching file python/mach/mach/telemetry.py
applying LibreWolf patch: patches/flatpak-autoconf.patch
patching file extensions/pref/autoconfig/src/nsReadConfig.cpp
applying LibreWolf patch: patches/fullpage-translations-customization.patch
patching file browser/base/content/main-popupset.inc.xhtml
patching file browser/base/content/main-popupset.js
patching file browser/components/preferences/main.inc.xhtml
patching file browser/components/preferences/main.js
patching file browser/components/translations/content/fullPageTranslationsPanel.js
patching file browser/locales/en-US/browser/translations.ftl
applying LibreWolf patch: patches/fullpage-translations.patch
patching file services/settings/Attachments.sys.mjs
Hunk #1 succeeded at 146 (offset 2 lines).
Hunk #2 succeeded at 206 (offset 2 lines).
Hunk #3 succeeded at 267 (offset 11 lines).
Hunk #4 succeeded at 438 (offset 11 lines).
Hunk #5 succeeded at 447 (offset 11 lines).
Hunk #6 succeeded at 500 (offset 11 lines).
Hunk #7 succeeded at 516 (offset 11 lines).
patching file services/settings/RemoteSettingsClient.sys.mjs
Hunk #4 succeeded at 610 (offset 1 line).
patching file toolkit/components/translations/actors/TranslationsParent.sys.mjs
Hunk #1 succeeded at 1217 (offset 183 lines).
Hunk #2 succeeded at 1605 (offset 183 lines).
Hunk #3 succeeded at 1685 (offset 209 lines).
Hunk #4 succeeded at 1711 (offset 209 lines).
Hunk #5 succeeded at 2006 (offset 337 lines).
applying LibreWolf patch: patches/hide-passwordmgr.patch
patching file browser/base/content/browser-init.js
patching file browser/components/preferences/privacy.js
Hunk #1 succeeded at 1274 (offset 60 lines).
applying LibreWolf patch: patches/librewolf-prefs.patch
patching file browser/app/profile/firefox.js
applying LibreWolf patch: patches/mozilla_dirs.patch
patching file toolkit/xre/nsXREDirProvider.cpp
Hunk #1 succeeded at 301 with fuzz 1 (offset 1 line).
Hunk #2 succeeded at 383 (offset 12 lines).
Hunk #3 succeeded at 415 (offset 4 lines).
Hunk #4 succeeded at 1134 (offset 20 lines).
Hunk #5 succeeded at 1400 (offset 37 lines).
Hunk #6 succeeded at 1410 (offset 37 lines).
applying LibreWolf patch: patches/msix.patch
patching file browser/installer/windows/msix/AppxManifest.xml.in
Hunk #1 succeeded at 30 (offset 1 line).
Hunk #2 succeeded at 120 (offset 11 lines).
patching file python/mozbuild/mozbuild/mach_commands.py
Hunk #1 succeeded at 2620 (offset 109 lines).
Hunk #2 succeeded at 2779 (offset 131 lines).
patching file python/mozbuild/mozbuild/repackaging/msix.py
Hunk #2 succeeded at 354 (offset -1 lines).
Hunk #3 succeeded at 480 (offset -29 lines).
applying LibreWolf patch: patches/nvidia-wayland-backported-fixes-0001-Bug-1898476-Wayland-Move-MozContainerSurfaceLock-fro.patch
The next patch would create the file widget/gtk/MozContainerSurfaceLock.cpp,
which already exists!  Assume -R? [n]
Apply anyway? [n]
Skipping patch.
1 out of 1 hunk ignored
The next patch would create the file widget/gtk/MozContainerSurfaceLock.h,
which already exists!  Assume -R? [n]
Apply anyway? [n]
Skipping patch.
1 out of 1 hunk ignored
patching file widget/gtk/MozContainerWayland.cpp
Reversed (or previously applied) patch detected!  Assume -R? [n]
Apply anyway? [n]
Skipping patch.
1 out of 1 hunk ignored -- saving rejects to file widget/gtk/MozContainerWayland.cpp.rej
patching file widget/gtk/MozContainerWayland.h
Reversed (or previously applied) patch detected!  Assume -R? [n]
Apply anyway? [n]
Skipping patch.
2 out of 2 hunks ignored -- saving rejects to file widget/gtk/MozContainerWayland.h.rej
patching file widget/gtk/moz.build
Reversed (or previously applied) patch detected!  Assume -R? [n]
Apply anyway? [n]
Skipping patch.
3 out of 3 hunks ignored -- saving rejects to file widget/gtk/moz.build.rej
```

### Notify maintainers
@ryand56 @dotlambda @squalus @mweinelt 

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.10.5, NixOS, 24.11 (Vicuna), 24.11.20240818.8a33541`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.5`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/d9gbq853jvbccrz5g3y0irbwgc57w137-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by GraithTiger** [comment]
Same issue here with home-manager on SteamOS.

I noticed that in https://github.com/NixOS/nixpkgs/issues/300107#issuecomment-2278374894 it was mentioned that the librewolf nix isn't pulling from the upstream source, but is simply patching firefox. Might this be related?

**Comment by ryand56** [comment]
This is caused because upstream is patching with the same patches introduced in #336204. Also, see #336304.

**Comment by MarkRijckenberg** [comment]
Same issue for me too.....

**Comment by P2949** [comment]
Same issue here

**Comment by amalgame21** [comment]
> This is caused because the package is applying the same patches introduced in #336204 that are also found in upstream. Also, see #336304.

Thanks for the pull request!
Just curious, can we apply this pull request with Override / OverrideAttrs before it merge in this case?
The package is a little bit complicated for me to do it because it is split into multiple file . I would appreciate if someone can guide us through it and help us learn more about nix language. Thanks!

**Comment by ryand56** [comment]
@amalgame21 Yes, just add my PR branch as an input to replace nixpkgs or copy the patch I made and override as such.

**Comment by squalus** [comment]
Fixed by #336304
